### PR TITLE
update test for mageck/test

### PR DESCRIPTION
**Pull request by famosab**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/mageck/test/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() },
+                { assert snapshot(
+                    process.out.gene_summary,
+                    process.out.sgrna_summary,
+                    process.out.r_script,
+                    process.out.versions
+                    )
```
[the remark this answers is not in the document]
Comment (sateeshperi) [reply modules/nf-core/mageck/test/tests/main.nf.test:36]: ya for outputs that dont generate consistent hashes...move them out of snapshot with `{ assert process.out. }`

File: modules/nf-core/mageck/test/tests/main.nf.test [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                 file(params.test_data['mus_musculus']['csv']['count_table'], checkIfExists: true)
+             ]
+                """
+            }
+        }
+
+                then {
+            assertAll(
+                { assert process.success },
+                { assert snapshot(process.out).match() },
```
Comment (mashehu) [inline modules/nf-core/mageck/test/tests/main.nf.test:30]:
```suggestion
```
if they are unstable as mentioned below, snapshotting the complete output will not work. See https://hackmd.io/@nf-core/ryWsFLwyT for example of alternative assertions for the unstable ones

**Review by maxulysse — APPROVED** [review]

**Comment by maxulysse** [comment]
oh, conda is not stable...

**Comment by SPPearce** [comment]
Looks like precision changes in the files, just check for a string or the first few lines.

**Comment by SPPearce** [comment]
Can you point me at the test you are struggling with?

File: modules/nf-core/mageck/test/tests/main.nf.test [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                            file(process.out.sgrna_summary.get(0).get(1)).readLines()[0],
+                            file(process.out.r_script.get(0).get(1)).readLines()[0]
+                            ).match()
+                }
+            )
+        }
+
+    }
+
+    test("mageck-test-day0 - csv") {
```
Comment (famosab) [inline modules/nf-core/mageck/test/tests/main.nf.test:41]: @SPPearce this test (day0) seems to not produce gene_summary or sgrna_summary output files

**Comment by mirpedrol** [comment]
Hi, I wonder if the output R script is needed, as it doesn't seem to be an output of the command. This is the inconsistent file when the tests are run with conda, so I think it would be easier to remove the `r_script` channel form the output and these the complete output with `{ assert snapshot(process.out).match() }`

**Comment by famosab** [comment]
Closed for #5819

**Pull request closed without merging**